tests/test_url.py

```diff
@@ -1235,11 +1235,29 @@ def test_with_path_query():
     assert str(url.with_path("/test")) == "http://example.com/test"
 
 
+def test_with_path_keep_query():
+    url = URL("http://example.com?a=b")
+    url2 = url.with_path("/test", with_query=True)
+    assert str(url2) == "http://example.com/test?a=b"
+
+
 def test_with_path_fragment():
     url = URL("http://example.com#frag")
     assert str(url.with_path("/test")) == "http://example.com/test"
 
 
+def test_with_path_keep_fragment():
+    url = URL("http://example.com#frag")
+    url2 = url.with_path("/test", with_fragment=True)
+    assert str(url2) == "http://example.com/test#frag"
+
+
+def test_with_path_keep_fragment_and_query():
+    url = URL("http://example.com?a=b#frag")
+    url2 = url.with_path("/test", with_query=True, with_fragment=True)
+    assert str(url2) == "http://example.com/test?a=b#frag"
+
```

Comment:
Please add tests for explicitly passing `keep_fragment=False` and `keep_query=False`